libpromises/pipes_unix.c

```diff
@@ -156,12 +156,25 @@ static pid_t GenericCreatePipeAndFork(IOPipe *pipes)
         return -1;
     }
 
+    /* WARNING: UNDEFINED BEHAVIOUR if the program is multi-threaded! */
     signal(SIGCHLD, SIG_DFL);
 
-    // Redmine #2971: reset SIGPIPE signal handler to have a sane behavior of piped commands within child
-    if (pid == 0)
+    if (pid == 0)                                               /* child */
     {
+        /* WARNING only call async-signal-safe functions in child. */
+
+        /* The fork()ed child is always single-threaded, but we are only
+         * allowed to call async-signal-safe functions (man 3p fork). */
+
+        // Redmine #2971: reset SIGPIPE signal handler in the child to have a
+        // sane behavior of piped commands within child
         signal(SIGPIPE, SIG_DFL);
+
+        /* The child should always accept all signals after it has exec'd,
+         * else the child might be unkillable! (happened in ENT-3147). */
+        sigset_t sigmask;
+        sigemptyset(&sigmask);
+        sigprocmask(SIG_SETMASK, &sigmask, NULL);
```

Comment:
Despite this intending to fix a bug in `cf-execd` (which has a block-all signal mask), the code here is in libpromises and it will affect all our binaries. I can't think of any bad side-effects though. @kacf can you?